### PR TITLE
avm2: Implement pixelBounds for transform

### PR DESCRIPTION
**Pull request by yoganlava**
Progresses: #10384
Code taken from avm1 transform

### CODE REVIEW

File: core/src/avm2/globals/flash/geom/transform.rs [hunk 1]
```diff
@@ -249,3 +249,29 @@ pub fn object_to_matrix<'gc>(
...
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let this = this.unwrap();
+    let display_object = get_display_object(this, activation)?;
+    rectangle_to_object(display_object.world_bounds(), activation)
+}
+
+pub fn rectangle_to_object<'gc>(
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/flash/geom/transform.rs:263]: Is there a reason for this function being public?

Comment (yoganlava) [reply core/src/avm2/globals/flash/geom/transform.rs:263]: Nope, copied from similar function up the file and forgot to remove, thanks

**Review by Herschel — APPROVED** [review]
I was going to suggest adding a test, but looks like we're missing `tests/avm2/displayobject_transform/Test.as` in the repo... 😆 In any case, the behavior is already covered by the AVM1 test and we can always add it if we'd like later.

**Comment by yoganlava** [comment]
Hmm seems to be failing for a test after tracing pixelBounds, put off the merge until I get to resolve it

**Comment by Herschel** [comment]
Good thing we added the test 😄  Thank you!

**Comment by yoganlava** [comment]
@Herschel Sorry haha, haven't resolved the issue yet. pixelBounds seems to be giving some wacky numbers but not sure where the issue lies

**Comment by Herschel** [comment]
> @Herschel Sorry haha, haven't resolved the issue yet. pixelBounds seems to be giving some wacky numbers but not sure where the issue lies

Oops, thx! Haven't looked into it yet, but how far off is it? Is it a precision issue? It's possible that the issue is in core `pixel_bounds` as I think it is somewhat naively implemented, in which case I'm okay with deferring that to a future PR.

**Comment by yoganlava** [comment]
I think it's due to stage bounds reporting the incorrect values.
e.g.
Flash
```
// stage.getBounds()
(x=0, y=0, w=0, h=0)
```
Ruffle:
```
// stage.getBounds()
(x=6710886.35, y=6710886.35, w=0, h=0)
```
I think the code in this PR is fine, as long as the bounds are fixed

**Comment by yoganlava** [comment]
I think i found the reason, the value reported by ruffle `6710886.35` * 20 = 134217727 = 0x7ffffff. The default value for a Rectangle<Twips> is 0x7ffffff or Twips::INVALID. I'm guessing flash sets the Rectangle bounds as 0 while we set them as 0x7ffffff

**Comment by Lord-McSweeney** [comment]
For the test I think you might need to add a `test.toml` and `output.txt` to actually run the test.

**Comment by yoganlava** [comment]
> For the test I think you might need to add a `test.toml` and `output.txt` to actually run the test.

Don't worry, I was just editing the existing test which still has the output and test.toml

**Review by Herschel — APPROVED** [review]
Thanks for the patience!

**Pull request merged**